client/authority-discovery/src/lib.rs

```diff
@@ -539,11 +540,18 @@ where
 
 		match inner() {
 			Ok(()) => {}
+
+			// Handle fatal errors.
+			//
+			// Given that the network likely terminated authority discovery should do the same.
+			Err(Error::DhtEventStreamTerminated) => return Poll::Ready(()),
```

Comment:
Why is the inner not returning a Poll?

Using a error to stop here sounds like a hack.

Reply:
I am not quite sure I follow.

Just to make sure my change is clear, this line terminates the authority discovery module for good. It is not a stream, thus the given task should not poll it again when this error happens.

The `inner` function could return a `Poll`, but I don't understand what use that would be. One would still need to inspect the error within and terminate the authority discovery in case the network terminates.

Would you mind adding some more details @bkchr?

Reply:
Discussed in Riot, decided to keep status quo.